File: Stacks/Patterns/Pattern2/Pattern2.py
stk=[]
lst=[1,5,2,11,6,4,5,11,10,100]
pge_lst=[0]*len(lst)

for i in range(0,len(lst)):
    while(len(stk)>0 and stk[-1]<=lst[i]):
        stk.pop()
    if len(stk)==0:
        pge_lst[i]=-1  
    else:
        pge_lst[i]=stk[-1]
    stk.append(lst[i])
print(pge_lst)

File: Stacks/Patterns/Pattern2/Pattern2.java
import java.util.*;
class Pattern2{
    public static void main(String[] args) {
        Stack<Integer> stk=new Stack<>();
        int[] num={5,2,7,3,6,3,2,10,66};
        int[] pge_num=new int[num.length];

        for (int i=0;i<num.length;i++){
            while(!stk.empty() && stk.peek()<=num[i]){
                stk.pop();
            }
            if(stk.empty()){
                pge_num[i]=-1;
            }
            else{
                pge_num[i]=stk.peek();
            }
            stk.push(num[i]);
        }
        System.out.println(Arrays.toString(pge_num));
    }
    
}
